### PR TITLE
Fix hardcode in githubRelease

### PR DESCRIPTION
**Pull request by Goooler**
## Description

Follow up #1730.

## Checklist

- [x] PR description added
- [ ] tests are added
- [ ] KtLint has been applied on source code itself and violations are fixed
- [ ] [documentation](https://pinterest.github.io/ktlint/) is updated
- [ ] `CHANGELOG.md` is updated

In case of adding a new rule:
- [ ] Rule is added to [rules documentation](https://pinterest.github.io/ktlint/rules/standard/)

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -61,18 +62,13 @@ val githubToken: String = if (project.hasProperty("servers.github.privKey")) {
...
-// Explicitly adding dependency on "shadowJarExecutable" as Gradle does not it set via "releaseAssets" property
 tasks.githubRelease {
-    dependsOn(provider { shadowJarExecutable })
-    dependsOn(provider { shadowJarExecutableChecksum })
+    setReleaseAssets(
+        provider {
+            projects.ktlint.dependencyProject.tasks.named<Checksum>("shadowJarExecutableChecksum")
+                .map { it.outputDirectory }
+        },
+    )
```
Comment (Goooler) [inline build.gradle.kts:71]: https://github.com/pinterest/ktlint/blob/20d9a8862187633e3336fb1d5abb8343df3b521e/ktlint/build.gradle.kts#L76-L84

**Comment by shashachu** [comment]
@Goooler I patched in this PR and I'm seeing similar behavior to when I tried to fix this last time, where it attempts to upload the directory:

```
:githubRelease UPLOADING ASSETS
:githubRelease UPLOADING /Users/shasha/code/ktlint/ktlint/build/run
java.io.FileNotFoundException: /Users/shasha/code/ktlint/ktlint/build/run (Is a directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at org.apache.tika.io.TikaInputStream.<init>(TikaInputStream.java:152)
        at org.apache.tika.io.TikaInputStream.get(TikaInputStream.java:382)
        at org.apache.tika.Tika.detect(Tika.java:288)
        at org.apache.tika.Tika$detect.call(Unknown Source)
```

You can try to test by setting `dryRun(true)` and running `clean githubRelease`; if it works it will list out all the files it needs to upload individually.

The other thing we could try is putting up a PR for the plugin so that it can handle directories as upload paths.

File: build.gradle.kts [hunk 2]
```diff
@@ -72,6 +72,7 @@ tasks.githubRelease {
 }
 
 githubRelease {
+    dryRun(true)
```
Comment (Goooler) [inline build.gradle.kts]: I can just see the 401 error after `dryRun`, which is the same as disabled

```
* What went wrong:
Execution failed for task ':githubRelease'.
> :githubRelease FAILED 401 
  [message:Bad credentials, documentation_url:https://docs.github.com/rest]
```

**Comment by shashachu** [comment]
Looks like this works!

```
:githubRelease [UPLOADING /Users/shasha/code/ktlint/ktlint/build/run/ktlint]
:githubRelease [UPLOADING /Users/shasha/code/ktlint/ktlint/build/run/ktlint.asc.md5]
:githubRelease [UPLOADING /Users/shasha/code/ktlint/ktlint/build/run/ktlint.asc]
:githubRelease [UPLOADING /Users/shasha/code/ktlint/ktlint/build/run/ktlint.md5]
```

**Comment by shashachu** [comment]
thank you @Goooler !

**Pull request merged**